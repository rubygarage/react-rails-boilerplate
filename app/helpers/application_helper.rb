module ApplicationHelper
  def destroy_admin_user_session_path
    "/log_out"
  end
end
