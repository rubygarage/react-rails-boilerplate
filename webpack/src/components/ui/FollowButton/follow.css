.card-btn {
  flex-grow: 1;
  flex-basis: 10rem;
  display: flex;
  color: white;
  align-items: center;
  text-align: left;

  &:first-of-type {
    padding-right: 1rem;
  }

  &:last-of-type {
    padding-left: 1rem;
  }

  &:hover {
    opacity: 0.6;
  }
}
