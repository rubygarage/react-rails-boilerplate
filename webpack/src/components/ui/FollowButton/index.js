import React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import classNames from 'classnames'
import Icon from 'components/ui/Icon'

import styles from './follow.css'

function FollowButton({ followersCount, followed, onFollow }) {
  return (
    <a href="" className={classNames(styles['card-btn'], 'card-btn-theme', 'profile-action')} onClick={onFollow}>
      <Icon type="icon-user-star-circle" className="font-30 mr-5" />
      <span>
        <span className="font-14 font-medium">
          <FormattedMessage id={`portfolio.${followed ? 'unfollow' : 'follow'}`} />
        </span>
        <br />
        <span className="sidebar-subtext-color">
          {followersCount}
        </span>
      </span>
    </a>
  )
}

FollowButton.propTypes = {
  onFollow: PropTypes.func,
  followed: PropTypes.bool,
  followersCount: PropTypes.number
}

export default FollowButton
