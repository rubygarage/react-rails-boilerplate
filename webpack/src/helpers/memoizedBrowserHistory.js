import createHistory from 'history/createBrowserHistory';

let instance = null;

export default class memoizedBrowserHistory {
  constructor() {
    if (!instance) {
      instance = createHistory();
    }

    return instance;
  }
}
