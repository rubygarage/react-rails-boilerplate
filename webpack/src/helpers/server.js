// moved the ugly environment (server/client) check to this helper
export const isServer = () => typeof window === 'undefined'
export const isClient = () => typeof window !== 'undefined'
