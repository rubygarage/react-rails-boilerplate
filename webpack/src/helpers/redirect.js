import memoizedBrowserHistory from 'helpers/memoizedBrowserHistory'

// uses browser router or server router depending on presence of response object in arguments
export const redirect = (path, res = null) => {
  if (res) {
    res.redirect(path)
    // we need to mark the response as finished to avoid double-render/double headers being set later in the stack
    // ( see src/server.js )
    res.finished = true
  } else {
    new memoizedBrowserHistory().push(path)
  }
}
