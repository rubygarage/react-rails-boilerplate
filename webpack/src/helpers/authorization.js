import { normalize } from 'normalize-json-api';
import { setTokenToStorage, deleteTokenFromStorage } from 'utils/tokens';
import { isClient } from 'helpers/server';
import ApiClient from 'utils/apiClient';

const authorizeUser = (req, resolve) => {
  const apiClient = new ApiClient().buildClient(req);

  apiClient.get('/api/v1/auth/users/session').then((response) => {
    const { entities, results } = normalize(response.data);

    if (isClient()) { setTokenToStorage(response.headers); }
    resolve({ currentUser: results, entities });
  }).catch(() => {
    deleteTokenFromStorage();
    resolve({ currentUser: {}, entities: {} });
  });
};

export default authorizeUser;
