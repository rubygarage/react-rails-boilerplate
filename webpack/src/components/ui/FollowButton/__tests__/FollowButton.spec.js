import React from 'react'
import { shallowWithIntl } from 'helpers/intl-enzyme'
import FollowButton from 'components/ui/FollowButton'

describe('<FollowButton />', () => {
  let props = {
    onFollow: jest.fn(),
    followed: true,
    followersCount: 77
  }

  it('renders followed button', () => {
    const component = shallowWithIntl(
      <FollowButton {...props} />
    )

    expect(component).toMatchSnapshot()
  })

  it('renders unfollowed button', () => {
    props = {
      ...props,
      followed: false
    }

    const component = shallowWithIntl(
      <FollowButton {...props} />
    )

    expect(component).toMatchSnapshot()
  })
})
