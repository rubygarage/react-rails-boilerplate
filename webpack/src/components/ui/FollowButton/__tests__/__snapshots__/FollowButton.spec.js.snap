// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FollowButton /> renders followed button 1`] = `
<a
  className="card-btn card-btn-theme profile-action"
  href=""
  onClick={[Function]}
>
  <Icon
    className="font-30 mr-5"
    type="icon-user-star-circle"
  />
  <span>
    <span
      className="font-14 font-medium"
    >
      <FormattedMessage
        id="portfolio.unfollow"
        values={Object {}}
      />
    </span>
    <br />
    <span
      className="sidebar-subtext-color"
    >
      77
    </span>
  </span>
</a>
`;

exports[`<FollowButton /> renders unfollowed button 1`] = `
<a
  className="card-btn card-btn-theme profile-action"
  href=""
  onClick={[Function]}
>
  <Icon
    className="font-30 mr-5"
    type="icon-user-star-circle"
  />
  <span>
    <span
      className="font-14 font-medium"
    >
      <FormattedMessage
        id="portfolio.follow"
        values={Object {}}
      />
    </span>
    <br />
    <span
      className="sidebar-subtext-color"
    >
      77
    </span>
  </span>
</a>
`;
